### PR TITLE
10.0 mig purchase request to requisition mgs

### PR DESCRIPTION
**Pull request by GillesTephaneMeyomesse**

### CODE REVIEW

File: purchase_request_to_requisition/models/purchase_requisition.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        return True
+
+    @api.multi
+    def action_in_progress(self):
+        res = super(PurchaseRequisition, self).action_in_progress()
+        self._purchase_request_confirm_message()
+        return res
+
+    @api.model
+    def _prepare_purchase_order_line(self, requisition, requisition_line,
```
Comment (adrienpeiffer) [inline purchase_request_to_requisition/models/purchase_requisition.py]: This method doesn't exist anymore

**Review by adrienpeiffer — CHANGES_REQUESTED** [review]

File: purchase_request_to_requisition/views/purchase_requisition_view.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0"?>
+<!-- Copyright 2017 Eficent Business and IT Consulting Services S.L.
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0) -->
+<odoo>
+    <data>
```
Comment (MiquelRForgeFlow) [inline purchase_request_to_requisition/views/purchase_requisition_view.xml]: remove data tags

**Review by MiquelRForgeFlow — CHANGES_REQUESTED** [review]

File: purchase_request_to_requisition/views/purchase_requisition_view.xml (hunk 2 above)
Comment (GillesTephaneMeyomesse) [reply purchase_request_to_requisition/views/purchase_requisition_view.xml]: @mreficent That is done thank's

Comment (MiquelRForgeFlow) [reply purchase_request_to_requisition/views/purchase_requisition_view.xml]: but you should do it in all the xml files, not only this one 😒😂

**Comment by AaronHForgeFlow** [comment]
Hey @GillesTephaneMeyomesse Thank you for migrate this module. Can you greenify this? :green_apple:

**Comment by GillesTephaneMeyomesse** [comment]
Hello @aheficent Of course, on my next commit

File: purchase_request_to_requisition/views/purchase_requisition_view.xml (hunk 2 above)
Comment (GillesTephaneMeyomesse) [reply purchase_request_to_requisition/views/purchase_requisition_view.xml]: Hi @mreficent Sorry, i will fix that on next commit

**Comment by GillesTephaneMeyomesse** [comment]
Hi @aheficent The travis errors come from unmet dependencie module (purchase_request_procurement). 
I will make small corrections on the purchase_request_procurement

**Comment by MiquelRForgeFlow** [comment]
This PR depends of https://github.com/OCA/purchase-workflow/pull/358.

**Comment by GillesTephaneMeyomesse** [comment]
yes and of #358 has little bugs that i'm trying to fix on

**Comment by AaronHForgeFlow** [comment]
This is merged https://github.com/OCA/purchase-workflow/pull/358 :clap:  Can you rebase this?

**Comment by GillesTephaneMeyomesse** [comment]
This branch was based i will correct any bug ASAP. Thks

**Comment by JordiBForgeFlow** [comment]
Ssems that Travis ia failing

**Comment by GillesTephaneMeyomesse** [comment]
@jbeficent I try to fix that. Thks

File: purchase_request_to_requisition/wizard/purchase_request_line_make_purchase_requisition_item.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                              required=True,
+                              readonly=True)
+    request_id = fields.Many2one('purchase.request',
+                                 related='line_id.request_id',
+                                 string='Purchase Request',
+                                 readonly=True)
+    product_id = fields.Many2one('product.product', string='Product')
+    name = fields.Char(string='Description', required=True)
+    product_qty = fields.Float(string='Quantity to Bid',
+                               digits=dp.get_precision('Product UoS'))
```
Comment (rousseldenis) [inline purchase_request_to_requisition/wizard/purchase_request_line_make_purchase_requisition_item.py:28]: @GillesTephaneMeyomesse I think this is 'Product Unit of Measure' instead. UoS is for sale purpose.

File: purchase_request_to_requisition/wizard/purchase_request_line_make_purchase_requisition_view.xml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                 <group>
+                    <field name="purchase_requisition_id"/>
+                 </group>
+                 <newline/>
+                 <group>
+                     <field name="item_ids" nolabel="1" colspan="2">
+                          <tree string="Details" create="false" editable="bottom">
+                              <field name="request_id"/>
+                              <field name="line_id"
+                                   options="{'no_open': true}"
```
Comment (rousseldenis) [inline purchase_request_to_requisition/wizard/purchase_request_line_make_purchase_requisition_view.xml:25]: @GillesTephaneMeyomesse This is not needed if invisible

File: purchase_request_to_requisition/wizard/purchase_request_line_make_purchase_requisition.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+class PurchaseRequestLineMakePurchaseRequisition(models.TransientModel):
+    _name = "purchase.request.line.make.purchase.requisition"
+    _description = "Purchase Request Line Make Purchase Requisition"
+
+    item_ids = fields.One2many(
+        'purchase.request.line.make.purchase.requisition.item',
+        'wiz_id', string='Items')
+    purchase_requisition_id = fields.Many2one(
+        'purchase.requisition', string='Purchase Requisition',
+        required=False, domain=[('state', '=', 'draft')])
```
Comment (rousseldenis) [inline purchase_request_to_requisition/wizard/purchase_request_line_make_purchase_requisition.py:19]: @GillesTephaneMeyomesse False is the default value

File: purchase_request_to_requisition/wizard/purchase_request_line_make_purchase_requisition.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        active_model = self.env.context.get('active_model')
+
+        if not request_line_ids:
+            return res
+        assert active_model == 'purchase.request.line', \
+            'Bad context propagation'
+
+        items = []
+        for line in request_line_obj.browse(request_line_ids):
+            items.append([0, 0, self._prepare_item(line)])
```
Comment (rousseldenis) [inline purchase_request_to_requisition/wizard/purchase_request_line_make_purchase_requisition.py:47]: @GillesTephaneMeyomesse Is this shouldn't be tuple ?

`[(0, 0, {...})]`

File: purchase_request_to_requisition/wizard/purchase_request_line_make_purchase_requisition.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+    def make_purchase_requisition(self):
+        pr_obj = self.env['purchase.requisition']
+        pr_line_obj = self.env['purchase.requisition.line']
+        company_id = False
+        picking_type_id = False
+        requisition = False
+        res = []
+        for item in self.item_ids:
+            line = item.line_id
+            if item.product_qty <= 0.0:
```
Comment (rousseldenis) [inline purchase_request_to_requisition/wizard/purchase_request_line_make_purchase_requisition.py:93]: @GillesTephaneMeyomesse You should use float_compare here

File: purchase_request_to_requisition/models/purchase_requisition_line.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    @api.multi
+    def action_openRequestLineTreeView(self):
+        """
+        :return dict: dictionary value for created view
+        """
+        request_line_ids = []
+        for line in self:
+            request_line_ids = [request_line.id for request_line
+                                in line.purchase_request_lines]
+        domain = [('id', 'in', request_line_ids)]
```
Comment (rousseldenis) [inline purchase_request_to_requisition/models/purchase_requisition_line.py:37]: @GillesTephaneMeyomesse You should have done :

`domain = [('id', 'in', self.mapped('purchase_request_lines.ids'))]`

File: purchase_request_to_requisition/models/purchase_request_line.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+from odoo import _
```
Comment (rousseldenis) [inline purchase_request_to_requisition/models/purchase_request_line.py:7]: @GillesTephaneMeyomesse Put it on following first import

File: purchase_request_to_requisition/models/purchase_requisition.py [hunk 10]
```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+from odoo import api, models
+from odoo import _
```
Comment (rousseldenis) [inline purchase_request_to_requisition/models/purchase_requisition.py:6]: @GillesTephaneMeyomesse Put this at the following of the first import

File: purchase_request_to_requisition/models/product.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+from odoo import api, models
+from odoo import _
```
Comment (rousseldenis) [inline purchase_request_to_requisition/models/product.py:6]: @GillesTephaneMeyomesse Pleas put it at the following of the first import

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Some minor comments

**Comment by MiquelRForgeFlow** [comment]
@GillesTephaneMeyomesse will you attend last comments? I think we should get this PR merged.

**Comment by AaronHForgeFlow** [comment]
Hi @GillesTephaneMeyomesse are you planning to attend the comments here? 

Otherwise, would you agree if we supersede this? 

Thanks

**Comment by MiquelRForgeFlow** [comment]
What do you think @rousseldenis ?

**Pull request closed without merging**